### DISCUSSION TITLE
Media Priors Update

### DISCUSSION THREAD

**Discussion by FedericoRaimondi** [google/lightweight_mmm#13] [Ideas]
Hello,

I think it would be a great addition to the repo to have a tutorial or some explanation on how to set up priors by media variable.
In the current documentation I see that:

> Media parameter beta_m is informed by costs. It uses a HalfNormal distribution and the scale of the distribution is the total cost of each media channel.

Including some additional prior knowledge to the model would be incredibly powerfull for such aggregated models.
Thanks in advance and keep up with this incredible work!

**Comment by pabloduque0** [comment]
Hello @FedericoRaimondi !

Thanks for opening this discussion and your kind words!

I think we can definitely add a full on tutorial/example on how the media priors work. In the meantime let me give you a few thoughts on it:

The current media priors are cost based, which means that we expect higher cost media to contribute more than lower cost one (one impression in an expensive channel would contribute more than one impression in cheap channel). See below an example of dummy data:

![image](https://user-images.githubusercontent.com/13575259/173098870-6ee00a51-7016-4855-a11c-4b55b0e3fdf6.png)

That said, if the HalfNormal distribution is a good fit for what you are aiming to do, you can provide values for the scale of such distribution that are unrelated to costs, in reality the model is agnostic to what unit/concept those priors relate to.

We are also working on giving more flexibility for the user to choose different priors for most of the parameters in the model, so that will come in soon :)

Let me know if there are any questions/comments left.

**Reply by FedericoRaimondi** [reply]
Thanks for your promt answer!

Overall I agree this is a good approximation given the assumption that we have same contracts rate with different media providers.
Probably we can encounter many cases where that assumption doesn't hold. Then, I guess your point of providing a different prior concept comes into place.

It is very good to see that you are planning to give more fexibility to the user. I might have some suggestions and hopefully you can share your thoughts on these as well:
- Flexibility to choose media coefficient priors
- Flexibility to choose media transforms priors (and possibility to set some parameters in advance, such as peak=0)
- Framing output for better usability (maybe there's already a way to get the estimated parameters in a tabular form)

And also a seperate question on the Carryover model:
Why did you choose the exponential transformation and not the hill transform like in [Jin, Y., Wang, Y., Sun, Y., Chan, D., & Koehler, J. (2017). Bayesian Methods for Media Mix Modeling with Carryover and Shape Effects. Google Inc.](https://research.google/pubs/pub46001/)?

**Reply by pabloduque0** [reply]
Yes I agree overall. Let me go through some of your points but please feel free to always do feature request, we are always interested in hearing what users are demanding (although will not be able to include things always) but definitely very interested in hearing your thoughts.

Our plan is that the user will have the ability to choose prior for any parameter in the model, so media coefficient and media transforms will be included in that.

Soon (ish) you will also be able to select the lagging and saturation separately, so you can make the combination that fits your case best.

What do you mean exactly by: `Framing output for better usability (maybe there's already a way to get the estimated parameters in a tabular form)` ?

**Reply by FedericoRaimondi** [reply]
Very excited to see the future developments! 

That point was to understand the possibility to have more model attributes such as coeff_ or similar.
It would facilitate contribution and due to calculation.

Thanks for the suggestion on feature request. Will definitely do as my time using the library increases.
